client_test.go

```diff
@@ -550,13 +552,19 @@ func sendPostRequest(t *testing.T, ctx context.Context, client *Client, url stri
 func expectSuccess(contents string) func(*testing.T, *http.Response, error) {
 	return func(t *testing.T, resp *http.Response, err error) {
 		t.Helper()
-		require.NoError(t, err)
+		if !assert.NoError(t, err) { //nolint:testifylint
+			return
+		}
```

Comment:
It's about calling `require` from a goroutine which is not the main go routine. Switched to `assert` to mark the test as failed but avoid corrupting it.

Reply:
That is really non-obvious, so this would greatly benefit from a comment about that.

In fact, you could hoist the entire `//nolint` to the func (so you don't need to repeat it for every `assert` call site) and put a comment there as to _why_ it's doing this.